src/main/kotlin/org/rust/ide/inspections/fixes/AttachFileToModuleFix.kt

```diff
@@ -164,16 +160,28 @@ private fun insertFileToModule(file: RsFile, mod: RsMod) {
     }
 
     WriteCommandAction.runWriteCommandAction(project) {
-        val child = mod.firstChild
-        val inserted = if (child == null) {
-            mod.add(modItem)
-        } else {
-            mod.addBefore(modItem, child)
-        } as RsModDeclItem
-        inserted.navigate(true)
+        insertModItem(modItem, mod).navigate(true)
     }
 }
 
+private fun insertModItem(item: RsModDeclItem, module: RsMod): RsModDeclItem {
+    val anchor = module.firstItem
+    val existingMod = generateSequence(anchor) {
+        it.getNextNonCommentSibling() as? RsElement
+    }
+        .filterIsInstance<RsModDeclItem>()
+        .lastOrNull()
+
+    val inserted = if (existingMod != null) {
+        module.addAfter(item, existingMod)
+    } else if (anchor != null) {
+        module.addBefore(item, anchor)
+    } else {
+        module.add(item)
+    }
```

Comment:
Let's use more idiomatic Kotlin style and replace `if`s with `when`
```suggestion
    val inserted = when {
        existingMod != null -> module.addAfter(item, existingMod)
        anchor != null -> module.addBefore(item, anchor)
        else -> module.add(item)
    }
```